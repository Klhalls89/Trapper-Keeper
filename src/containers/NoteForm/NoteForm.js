import React, { Component } from 'react';
import  PropTypes from 'prop-types';
import { connect } from 'react-redux';

class NoteForm extends Component {
  constructor() {
    super();
    this.state = {
      title: '',
      listItem: '',
      list: []
    }
  }

  handleChange = (e) => {
    const { name, value } = e.target;
    
    this.setState({
      [name]: value
    });
  }

  updateList = (e) => {
    const newItem = e.target.value;

    this.setState({
      listItem: '',
      list: [...this.state.list, newItem]
    });
    // this.props.getNotes(this.state.list)
  }

  handleKeyPress = (e) => {
    if (e.key === 'Enter') {
      this.updateList(e);
    }
  }

  render() {
    const itemInput = 
      <input 
        type="text"
        placeholder="List item"
        name="listItem"
        onChange={this.handleChange}
        onKeyPress={this.handleKeyPress}
        onBlur={this.updateList} />

    return (
      <section>
        <input 
          type="text" 
          placeholder="Title"
          name="title"
          value={this.state.title}
          onChange={this.handleChange} />
        {itemInput}
      </section>
    );
  }
}

export const mapStateToProps = ({notes}) => ({
  notes
})

export const mapDispatchToProps = dispatch => ({
  getNotes: list => dispatch(getNotes(list))
})



export default connect(mapStateToProps, mapDispatchToProps)(NoteForm);

NoteForm.propTypes = {
  title: PropTypes.string,
  list: PropTypes.array
}

