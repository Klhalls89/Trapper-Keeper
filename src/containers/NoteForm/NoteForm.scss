.note-form-bg {
  background: rgba(#DCDCDC, .6);
  display: flex;
  height: 120%;
  justify-content: center;
  position: absolute;
  top: 0;
  width: 100%;
  z-index: 2;
}

.note-form {
  background: white;
  border-radius: 5px;
  box-shadow: 1px 2px 5px #666;
  display: flex;
  flex-direction: column;
  padding: 15px;
  position: absolute;
  top: 200px;
  width: 600px;

  input,
  button {
    background: transparent;
    border: 0;
    line-height: 1.25rem;
  }

  .note-title {
    color: $black;
    font-size: 1.3em;
    padding: 5px 0 10px 0;
  }

  hr {
    border: .5px solid;
    color: #888;
  }
}

.fa-plus {
  color: $dkGrey
}

.hide {
  display: none;
}

.list-items-wrapper {
  padding: 0 15px;
}

.new-item-input {
  .fa-plus {
    color: #888;
    margin: 0 8px 0 0;
  }

  input {
    outline: none;
    padding: 7px 0;
  }
}

.form-btn-wrapper {
  display: flex;
  justify-content: space-between;
  margin: 20px 0 0 0;
}

.save-btn,
.fa-trash-alt,
.fa-palette {
  color: #888;
  cursor: pointer;
  padding: 5px 0;

  &:hover {
    color: $black;
  }
}
