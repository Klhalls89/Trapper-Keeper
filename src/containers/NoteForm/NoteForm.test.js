import React from 'react';
import { shallow } from 'enzyme';
import NoteForm from './NoteForm';

describe('NoteForm', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <NoteForm />
    );
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a default state', () => {
    expect(wrapper.state()).toEqual({
      title: '',
      listItem: '',
      list: []
    });
  });

  it('should update title or list item with keypress', () => {
    expect(wrapper.state('title')).toEqual('');

    const mockEvent = {
      target: { name: 'title', value: 'this is a title' }
    }

    wrapper.instance().handleChange(mockEvent);

    expect(wrapper.state('title')).toEqual('this is a title');
  });

  it('should update list with item input', () => {
    expect(wrapper.state('listItem')).toEqual('')

    const mockEvent = { 
      key: 'Enter', 
      target: { value: 'this is a list item' }
    }

    wrapper.instance().handleKeyPress(mockEvent);

    expect(wrapper.state('list')).toEqual(['this is a list item'])
  });
});