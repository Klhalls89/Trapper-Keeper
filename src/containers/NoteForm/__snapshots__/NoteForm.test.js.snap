// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteForm should render correctly 1`] = `undefined`;
